openmdao/components/exec_comp.py

```diff
@@ -222,28 +205,110 @@ def __init__(self, exprs=[], **kwargs):
         self._codes = None
         self._kwargs = kwargs
 
+        self._manual_decl_partials = False
         self._no_check_partials = True
 
+    def initialize(self):
+        """
+        Declare options.
+        """
+        self.options.declare('has_diag_partials', types=bool, default=False,
+                             desc='If True, treat all array/array partials as diagonal if both '
+                                  'arrays have size > 1. All arrays with size > 1 must have the '
+                                  'same flattened size or an exception will be raised.')
+
+        self.options.declare('units', types=str, allow_none=True, default=None,
+                             desc='Units to be assigned to all variables in this component. '
+                                  'Default is None, which means units may be provided for variables'
+                                  ' individually.',
+                             check_valid=check_option)
+
+        self.options.declare('shape', types=(int, tuple, list), allow_none=True, default=None,
+                             desc='Shape to be assigned to all variables in this component. '
+                                  'Default is None, which means shape may be provided for variables'
+                                  ' individually.')
+
+        self.options.declare('shape_by_conn', types=bool, default=False,
+                             desc='If True, shape all inputs and outputs based on their '
+                                  'connection. Default is False.')
+
+    @classmethod
+    def register(cls, name, callable_obj, complex_safe):
+        """
+        Register a callable to be used within an ExecComp expression.
```

Comment:
maybe reword to something like: "Register a callable that can be used within all ExecComp expressions."

Reply:
reworded